### ISSUE TITLE
Is it possible to use EIP-192 precompile?

### ISSUE THREAD

**Issue by k06a** [kilic/evmvdf#1]
Is it possible to use EIP-192 precompiled smart contract to make verification even cheaper? https://github.com/ethereum/EIPs/blob/master/EIPS/eip-198.md

>Rationale
>
>This allows for efficient RSA verification inside of the EVM, as well as other forms of number theory-based cryptography. Note that adding precompiles for addition and subtraction is not required, as the in-EVM algorithm is efficient enough, and multiplication can be done through this precompile via a * b = ((a + b)**2 - (a - b)**2) / 4.
>
>The bit-based exponent calculation is done specifically to fairly charge for the often-used exponents of 2 (for multiplication) and 3 and 65537 (for RSA verification).

**Comment by kilic** [comment]
It is already in use for modular exponentiations see [here](https://github.com/kilic/evmvdf/blob/71c0d45b9f26bfbf1393b63adc850535bb10f3c6/contracts/VerifyVDF.sol#L163) and [here](https://github.com/kilic/evmvdf/blob/71c0d45b9f26bfbf1393b63adc850535bb10f3c6/contracts/VerifyVDF.sol#L139)

**Comment by k06a** [comment]
@kilic I meant it could be used for `mul2048()`

**Comment by kilic** [comment]
@k06a Yes It can be used for multiplication with `ab = ((a+b)^2 - (a-b)^2) >> 2`. Let's see whether it reduces the cost.

**Comment by k06a** [comment]
@kilic also you could try to use `uint256[8]` instead of `bytes` it is stored similar way but without length prefix.
